File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/response/AssignmentGradingResponse.java
package com.hoangphihiep.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AssignmentGradingResponse {

    private String studentId;

    private String studentName;

    private String email;

    private Integer totalAssignments;

    private Integer submittedAssignments;

    private Integer gradedAssignments;

    private Integer pendingAssignments;

    private Integer averageScore;

    private List<AssignmentSubmissionResponse> latestSubmissions;
}

File: react-type-vite/src/services/api/response/assignmentGradingResponse.ts
import type { AssignmentSubmissionResponse } from "./assignmentSubmissionResponse"

export interface AssignmentGradingResponse {
  studentId: string
  studentName: string
  email: string
  totalAssignments: number
  submittedAssignments: number
  gradedAssignments: number
  pendingAssignments: number
  averageScore: number
  latestSubmissions: AssignmentSubmissionResponse[]
}